src/network/networkd-link.c

```diff
@@ -284,6 +284,29 @@ static int link_enable_ipv6(Link *link) {
         return 0;
 }
 
+static int link_disable_ipv6_addr_gen_mode(Link *link) {
+        const char *p = NULL;
+        int r;
+
+        /* Make this a NOP if IPv6 is not available */
+        if (!socket_ipv6_is_supported())
+                return 0;
+
+        if (link->flags & IFF_LOOPBACK)
+                return 0;
+
+        if (link_ipv6ll_enabled(link))
+                return 0;
+
+        p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/addr_gen_mode");
+
+        r = write_string_file(p, "1", WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
In networkd-link.c, we usually also set `WRITE_STRING_FILE_VERIFY_ON_FAILURE`.

Reply:
updated thanks !